File: src/main/kotlin/predcompiler/compilation/evaluation/StateRobustnessInfo.java
package predcompiler.compilation.evaluation;

public class StateRobustnessInfo {

	/**
	 * Robustness of the self loop transition, if any. Defaults to
	 * Float.NEGATIVE_INFINITY if no self loop exists.
	 */
	public float selfLoopRobustness;

	/**
	 * Information on the edge with the highest robustness of the outgoing edges.
	 */
	public EdgeRobustnessInfo bestOutgoingEdgeData;

} // StateRobustnessInfo

File: src/main/kotlin/predcompiler/compilation/evaluation/StateRobustnessInfo.kt
package predcompiler.compilation.evaluation

data class StateRobustnessInfo (
    /**
     * Robustness of the self loop transition, if any. Defaults to
     * Float.NEGATIVE_INFINITY if no self loop exists.
     */
    val selfLoopRobustness: Double = 0.0,

    /**
     * Information on the edge with the highest robustness of the outgoing edges.
     */
    val bestOutgoingEdgeData: EdgeRobustnessInfo? = null
) // StateRobustnessInfo
